### PR TITLE
Java 10 Integration

### PR DESCRIPTION
**Pull request by cdietrich**
use asm 6 and jdt Oxygen 3a
Fixed changed ASM Behaviour with 6.1.x
adjusted pipelines

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.common.types.tests/META-INF/MANIFEST.MF_gen [hunk 1]
```diff
@@ -6,7 +6,7 @@ Bundle-Version: 0.0.1
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Bundle-ActivationPolicy: lazy
 Export-Package: org.eclipse.xtext.common.types.xtext.ui
-Require-Bundle: org.objectweb.asm;bundle-version="[5.0.1,6.0.0)";resolution:=optional,
+Require-Bundle: org.objectweb.asm;bundle-version="[6.1.1,7.0.0)";resolution:=optional,
```
Comment (holgerschill) [inline org.eclipse.xtext.common.types.tests/META-INF/MANIFEST.MF_gen]: Why upper bound to 7.0.0?

File: org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/types/TypesGeneratorFragment.java [hunk 2]
```diff
@@ -44,12 +44,12 @@
 
 	@Override
 	public String[] getRequiredBundlesRt(Grammar grammar) {
-		return new String[] { "org.eclipse.xtext.common.types","org.objectweb.asm;bundle-version=\"[5.0.1,6.0.0)\";resolution:=optional" };
+		return new String[] { "org.eclipse.xtext.common.types","org.objectweb.asm;bundle-version=\"[6.1.1,7.0.0)\";resolution:=optional" };
```
Comment (holgerschill) [inline org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/types/TypesGeneratorFragment.java]: Why upper bound to 7.0.0?

Comment (cdietrich) [reply org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/types/TypesGeneratorFragment.java]: cause we had always had  generous range here and i did not want to change it

File: org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java [hunk 3]
```diff
@@ -3890,4 +3866,34 @@ public void testBug470767() {
 		JvmTypeReference paramType = method.getParameters().get(0).getParameterType();
 		assertEquals("int", paramType.getSimpleName());
 	}
+
+	@Test
+	public void testInnerAnnotationType() throws Exception {
```
Comment (spoenemann) [inline org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: What is the cause for the change to this test?

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: asm fixed a bug/bad behaviour. thus the change/additional test cases + the fix

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java]: yes

**Comment by cdietrich** [comment]
@spoenemann removed the TODO

**Pull request merged**